lambdas/post_to_slack/post_to_slack.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+# -*- encoding: utf-8 -*-
+"""
+Sends slack notifications for alarms events
+"""
+
+import json
+import pprint
+import os
+
+from botocore.vendored import requests
+
+
+def main(event, _):
+    print(f'Received event:\n{pprint.pformat(event)}')
+    message = event['Records'][0]['Sns']['Message']
+    message_data = json.loads(message)
+    alarm_name = message_data['AlarmName']
+    namespace = message_data['Trigger']['Namespace']
+    metric_name = message_data['Trigger']['MetricName']
+    dimensions = message_data['Trigger']['Dimensions']
+    state_reason = message_data['NewStateReason']
+
+    slack_data = {'username': 'cloudwatch-alert',
+                  "icon_emoji": ":slack:",
+                  "attachments": [{
+                      'color': 'danger',
+                      'fallback': alarm_name,
+                      "title": alarm_name,
+                      "fields": [
+                          {
+                              "title": "Metric",
+                              "value": f"{namespace}/{metric_name}"
+                          },
+                          {
+                              "title": "Dimensions",
+                              "value": f"{pprint.pformat(dimensions)}"
+                          },
+                          {
+                              "title": "Reason",
+                              "value": state_reason
+                          }
+                      ]
+                  }]}
+
+    response = requests.post(
+        os.environ["SLACK_INCOMING_WEBHOOK"],
+        data=json.dumps(slack_data),
+        headers={'Content-Type': 'application/json'}
+    )
+    if response.status_code != 200:
```

Comment:
Neat, but can be neater still: Response objects have a `raise_for_status()` method you can use instead:

```pycon
>>> import requests
>>> r = requests.get('https://api.wellcomecollection.org/doesnotexist')
>>> r
<Response [404]>
>>> r.raise_for_status()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/chana/.virtualenvs/api/lib/python3.6/site-packages/requests/models.py", line 909, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://api.wellcomecollection.org/doesnotexist
```